### PR TITLE
feat(argocd-apps): support ignoreApplicationDifferences for appsets

### PR DESCRIPTION
**Pull request by imranismail**
Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

File: charts/argocd-apps/values.yaml [hunk 1]
```diff
@@ -97,6 +97,10 @@ applicationsets: []
 #   namespace: argocd
 #   additionalLabels: {}
 #   additionalAnnotations: {}
+#   # Ref https://argo-cd.readthedocs.io/en/stable/operator-manual/applicationset/Controlling-Resource-Modification/#allow-temporarily-toggling-auto-sync
+#   ignoreAppDifferences:
+#   - jsonPointers:
+#     - /spec/syncPolicy
```
Comment (yu-croco) [inline charts/argocd-apps/values.yaml:103]: Can you please add the attributes in CI as well ? 
https://github.com/argoproj/argo-helm/blob/main/charts/argocd-apps/ci/applicationsets-values.yaml

File: charts/argocd-apps/Chart.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 name: argocd-apps
 description: A Helm chart for managing additional Argo CD Applications and Projects
 type: application
-version: 1.4.1
+version: 1.5.0
 home: https://github.com/argoproj/argo-helm
 icon: https://argo-cd.readthedocs.io/en/stable/assets/logo.png
 keywords:
```
Comment (yu-croco) [inline charts/argocd-apps/Chart.yaml:8]: > I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).

Please also update (remove old one and add new one ) at `artifacthub.io/changes:` block as well.

**Review by yu-croco — CHANGES_REQUESTED** [review]
Thank you for your PR. :)
 I added some small requests.

File: charts/argocd-apps/Chart.yaml (hunk 2 above)
Comment (imranismail) [reply charts/argocd-apps/Chart.yaml:8]: Added

**Comment by imranismail** [comment]
@yu-croco thank you for the prompt review, added the requested changes

**Review by MattRipia — APPROVED** [review]

**Comment by imranismail** [comment]
This is a puzzling. The test is supposed to check the functionality of the argocd-apps chart, but it fails on some other charts that are not directly related.

**Comment by yu-croco** [comment]
> Installing chart with values file "charts/argocd-apps/ci/extensions-values.yaml"...
>
> Creating namespace "argocd-apps-re9ctpz5xl"...
namespace/argocd-apps-re9ctpz5xl created
Error: INSTALLATION FAILED: unable to build kubernetes objects from release manifest: resource mapping not found for name: "example" namespace: "default" from "": no matches for kind "ArgoCDExtension" in version "argoproj.io/v1alpha1"
ensure CRDs are installed first

[argocd-apps/ci/extensions-values.yaml](https://github.com/argoproj/argo-helm/blob/main/charts/argocd-apps/ci/extensions-values.yaml) has problem... 👀 
CI Ref: https://github.com/argoproj/argo-helm/actions/runs/7618206138/job/20753238610?pr=2442

Since Argo CD Extensions is [deprecated](https://github.com/argoproj-labs/argocd-extensions?tab=readme-ov-file#deprecation-notice) now, I wonder we don't need support this so eagerly, which means we can remove [argocd-apps/ci/extensions-values.yaml](https://github.com/argoproj/argo-helm/blob/main/charts/argocd-apps/ci/extensions-values.yaml)...? 🤔 
 
@mkilchhofer @jmeridth 
What do you think about this? 🤔

**Comment by yu-croco** [comment]
I made PR for failed CI, https://github.com/argoproj/argo-helm/pull/2449 . 🙋

**Comment by yu-croco** [comment]
Hi @imranismail ,
please fetch latest diff and you can pass the CI. :)

File: charts/argocd-apps/Chart.yaml [hunk 3]
```diff
@@ -17,5 +17,5 @@ annotations:
     fingerprint: 2B8F22F57260EFA67BE1C5824B11F800CD9D2252
     url: https://argoproj.github.io/argo-helm/pgp_keys.asc
   artifacthub.io/changes: |
-    - kind: deprecated
-      description: Add deprecated note for Argo CD Extensions
+    - kind: added
+      description: ability to specify ignoreApplicationDifferences for applicationsets
```
Comment (yu-croco) [inline charts/argocd-apps/Chart.yaml:21]: You need new line in the end to pass CI lint 🙋 

https://github.com/argoproj/argo-helm/pull/2442/files/dc324fd474e2c785f25dc9ce1bd49b270b54189f..a53d032ae8f7974701a8d41b8239853691f1489f

Comment (yu-croco) [reply charts/argocd-apps/Chart.yaml:21]: Also please bump version, maybe you missied it when you fetch remote latest diff.

**Review by yu-croco — APPROVED** [review]
Thank you for your contribution. LGTM

**Review by mbevc1 — APPROVED** [review]

**Pull request merged**